watchdog.go

```diff
@@ -364,11 +373,12 @@ func pollingWatchdog(policy Policy, frequency time.Duration, limit uint64, usage
 // forceGC forces a manual GC.
 func forceGC(memstats *runtime.MemStats) {
 	Logger.Infof("watchdog is forcing GC")
+	notifyGC(GCTriggerWatchdog)
```

Comment:
I think this will trigger two notifications: the one we're emitting here, AND the one that is emitted after GC runs. That's certainly not what we want.

Reply:
Yes, that's intentional. Otherwise watchdog triggers the GC run, _and then_ the notification is triggered, meaning that the state freed up in response to the trigger is not gc'ed until the _next_ GC run.

Do we expect applications care about regular GC runs at all? In the connmgr, I'm ignoring any Go-triggered GC run, I'm _only_ interested in watchdog-triggered runs. Maybe we should change the meaning of `RegisterNotifee` to just watchdog-triggered runs?